ext/json/json_parser.tab.c

```diff
@@ -1882,12 +1997,17 @@ int php_json_parser_object_update(php_json_parser *parser, zval *object, zend_st
 		ZVAL_NEW_STR(&zkey, key);
 		zend_std_write_property(object, &zkey, zvalue, NULL); 
 
-		if (Z_REFCOUNTED_P(zvalue)) {
-			Z_DELREF_P(zvalue);
-		}
+		zval_dtor(zvalue);
 	}
 	zend_string_release(key);
 
+	if (EG(exception)) {
+		parser->scanner.errcode = PHP_JSON_ERROR_DEPTH;
```

Comment:
I think that it would be useful for users that want to identify that there is some attempt to attack their app. Lets say you have got a public API and you would like to identify that someone is sending such json (possibly trying to dos php version without this patch), then you might want to like to possibly log such case specially. That was my thinking when I suggested a new error... :)